packages/react/src/hooks/usePreservedCallback.ts

```diff
@@ -1,14 +1,14 @@
 import { useCallback, useRef } from 'react'
 import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect'
 
-export const usePreservedCallback = <T extends (...args: unknown[]) => unknown>(callback: T) => {
+export const usePreservedCallback = <T extends (...args: any[]) => unknown>(callback: T): T => {
   const callbackRef = useRef<T>(callback)
 
   useIsomorphicLayoutEffect(() => {
     callbackRef.current = callback
   }, [callback])
 
-  return useCallback((...args: unknown[]) => {
+  return useCallback((...args: Parameters<T>) => {
     return callbackRef.current(...args)
   }, []) as T
 }
```

Comment:
Could you add type test with usePreservedCallback.test-d.ts?

Reply:
Sure! I added it.